pkg/controller/controller_test.go

```diff
@@ -376,3 +373,46 @@ func TestIsValidOneOf(t *testing.T) {
 		})
 	}
 }
+
+func TestGetEmail(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name    string
+		email   string
+		wantRes string
+		wantErr string
+	}{
+		{
+			name:    "empty_email",
+			email:   "",
+			wantErr: "email header is not valid",
+		},
+		{
+			name:    "incorrect_format_email",
+			email:   "iap-prefix/test@email.com",
+			wantErr: "email value has unexpected format",
+		},
+		{
+			name:    "happy_path1",
+			email:   iapHeaderName + ":test@email.com",
+			wantRes: "test@email.com",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			gotRes, err := getEmail(tc.email)
+			if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
+				t.Errorf("Unexpected err: %s", diff)
+			}
+			if diff := cmp.Diff(tc.wantRes, gotRes); diff != "" {
```

Comment:
Use `!=` for primitive type comparison

Reply:
Im probably overthinking this, could you provide an example? Im thinking if I do != then I wont have a useful error message to print.

Reply:
You just write it yourself

```go
if tc.wantRes != gotRest {
  t.Errorf("email got=%s want=%s", gotRes, tc.wantRes)
}
```

Reply:
Even better, shadow the variable to make the error message and format consistent:

```go
if got, want := gotRest, tc.wantRes; got != want {
  t.Errorf("expected %q to be %q", got, want)
}
```